lib/rubocop/cop/rspec/expect_output.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for opportunities to use `expect { ... }.to output`.
+      #
+      # @example
+      #   # bad
+      #   $stdout = StringIO.new
+      #   my_app.print_report
+      #   $stdout = STDOUT
+      #   expect($stdout.string).to eq('Hello World')
+      #
+      #   # good
+      #   expect { my_app.print_report }.to output('Hello World').to_stdout
+      class ExpectOutput < Cop
+        MSG = 'Use `expect { ... }.to output(...).to_%<name>s` ' \
+              'instead of mutating $%<name>s'.freeze
+
+        def_node_matcher :example?, Examples::ALL.block_pattern
+        def_node_matcher :hook?, Hooks::ALL.block_pattern
+
+        def on_gvasgn(node)
+          return unless inside_example_scope?(node)
+
+          variable_name, = *node
```

Comment:
I really like when we assign to underscore-prefixed variables. When writing new cops, it is nice to be able to look at the existing ones and see what kind of arguments are passed into the various `on_` methods.